src/moonlink/src/table_handler.rs

```diff
@@ -31,6 +31,150 @@ pub struct EventSyncSender {
     pub flush_lsn_tx: watch::Sender<u64>,
 }
 
+#[derive(PartialEq)]
+pub enum SpecialTableState {
```

Comment:
curious will it be used outside? 
i.e. I would imagine it to be "mooncake table" or "moonlink" internal thing, at most it should be `pub(crate)` rather than `pub`.